### ISSUE TITLE
Not possible to register WebApp name/key properly when using WebRequestTrackingFilter in Spring boot 

### ISSUE THREAD

**Issue by gdahlin** [Microsoft/ApplicationInsights-Java#359]
Hi,

When using the WebRequestTrackingFilter with a Spring boot application on the root context path the filter cannot register the web app name/key properly:

```
AI: ERROR 30-01-2017 12:15, 22: WebApp name is not found, unable to register WebApp
AI: INFO 30-01-2017 12:15, 22: Successfully registered the filter 'ApplicationInsightsWebFilter
```

This seem to be due to dependencies to either having a (non-root) servlet context path set for the app or that the app is found under a class-loading path containing "WEB-INF", which is not the case for this Spring boot app.

Would be good if the initialisation config could read either the spring boot app name or to have some external configuration value to set for the web app name/key.

Thanks in advance

**Comment by tomaszglinski** [comment]
I managed to fix that by implementing wrappers for `WebRequestTrackingFilter`, `FilterContext `and `ServletContext `classes, but that was a dirty hack...
Strange that such parameter has not been added from the very beginning...

**Comment by ericis** [comment]
Yes, I'm getting the same error. It's not an enhancement IMO. Seems like a bug.

**Comment by twsouthwick** [comment]
@tomaszglinski What was the work around that you did?

**Comment by tomaszglinski** [comment]
@twsouthwick I think I extended the mentioned classes to pass required parameter. Cannot say exactly, cause I removed that (I was only playing, I resigned from App insights because of those troubles)

**Comment by dhaval24** [comment]
fixed in #449 with a public constructor to register the name of spring boot application. Later we will take the approach of #461 for better SpringBoot Integration.
